coco/master.py

```diff
@@ -59,6 +75,46 @@
     return response.raw(result, headers={"Content-Type": "application/json"})
 
 
+# Prometheus endpoint
+# This server can only export the total requests. Individual success/fail
+# counts for every host will be exported separately by the forwarder process
+@app.get("/metrics")
+async def metrics(request):
+    """
+    Prometheus metrics endpoint.
+    """
+    try:
+        output = generate_latest(REGISTRY).decode("utf-8")
+        content_type = CONTENT_TYPE_LATEST
+        return response.text(body=output, content_type=content_type)
+    except Exception as e:
+        msg = f"{e}"
+        logger.error(msg)
+        raise ServerError(msg)
+
+
+@app.listener('after_server_stop')
+async def cleanup_prometheus(app, loop):
+    """
+    Clean up prometheus metric files on exit.
+    """
+    multiprocess.mark_process_dead(os.getpid())
+
+
+@app.listener('before_server_start')
+async def init_metrics(app, loop):
+    """
+    Set up counters for every endpoint.
+    """
+    logger.error(f"{ENDPOINTS}")
+    global COUNTERS
+    for edpt in ENDPOINTS:
+        COUNTERS[edpt] = Counter(format_metric_name(f"coco_{edpt}_dropped"),
```

Comment:
This should get a unit (pass it a `unit="total"`)

Reply:
And I think you can pass `"coco"` as a prefix or similar. What's the regex magic for?

Reply:
in this case all the formatting does is replace '-' with '_'. The regex is to separate the hostname and the port from the URL for the labels.

Reply:
That makes sense